### PR TITLE
fix(auth): Added hash param to cognito api methods

### PR DESCRIPTION
**Pull request by banji180**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [x] Added Unit Tests
- [x] Added Integration Tests

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/FetchAuthSessionTestCaseGenerator.kt [hunk 1]
```diff
@@ -1,3 +1,18 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (gpanshu) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/FetchAuthSessionTestCaseGenerator.kt:2]: Change to 2023. Do that for all files

**Review by gpanshu — DISMISSED** [review]

**Review by gpanshu — APPROVED** [review]

File: aws-auth-cognito/src/test/java/featureTest/utilities/APICaptorFactory.kt [hunk 2]
```diff
@@ -105,6 +107,26 @@ class APICaptorFactory(
                 every { consumer.call() } answers { latch.countDown() }
                 successCaptors[apiName] = actionCaptor
             }
+            AuthAPI.rememberDevice -> {
```
Comment (div5yesh) [inline aws-auth-cognito/src/test/java/featureTest/utilities/APICaptorFactory.kt:110]: Are these changes related to this PR? Please rebase.
